### ISSUE TITLE
not getting callback from websocket after sending some nano from an external wallet to a NanoClient wallet address

### ISSUE THREAD

**Issue by OmarB97** [nanobox-cc/nano-client#9]
When using the new websocket option for the nanoClient and adding onSend and onReceive listeners, and sending nano from an external wallet to a newly generated NanoClient wallet address, I am not getting the onSend or onReceive callback.

**Comment by mehlium** [comment]
Hey! Can you try again? There seems to have been an issue with the websockets configuration.

**Comment by mehlium** [comment]
Also, keep in mind that onRecieve listener will happen first when a receive is broadcasted, not when a send event happens. This makes it hard to listen for "receives" from unknown accounts. I'll add it as a feature request.

**Comment by OmarB97** [comment]
> Also, keep in mind that onRecieve listener will happen first when a receive is broadcasted, not when a send event happens. This makes it hard to listen for "receives" from unknown accounts. I'll add it as a feature request.

Yeah that was my suspicion. In my custom websocket implementation I was able to get it to subscribe to send blocks which were sending nano from an unknown account to my account. Would be really nice to have that feature as it's quite crucial for my app's functionality :) Until then will probably need to stick to my own websocket implementation

**Comment by mehlium** [comment]
Yeah, found an error in the websocket implementation on my end. Will have out a fix tomorrow!

**Comment by mehlium** [comment]
@OmarB97 took your inspiration and implemented only using send blocks in #12, meaning:

When a sendblock is broadcasted:

*  `onSend` will trigger for the account the block was sent from
* `onReceive` will trigger for the linked account the block was sent to

This means that to actually "pocket" the funds, the client needs to call `.receice(account)` afterwards, which I think is fine given how Nano works. Happy to hear your thoughts on how you use this in your implementation. :-)

**Comment by OmarB97** [comment]
Nice! Still concerning though since now there’s seems to be no way to
listen for the receive confirmation block after calling .receive() if I’m
understanding correctly.

**Comment by mehlium** [comment]
That's correct. In what cases would you need to listen for the `receive` events?

**Comment by OmarB97** [comment]
Hi, sorry for the late response, but generally I would find it useful to
get the receive confirmation subscription to know for certain that the
received nano is no longer pending on the account.

One other question: are there issues with the NanoBox node right now? I'm
trying to do send/receive calls and it seems like none are going through at
the moment (since roughly 1.5 hours ago during my testing). Is the node
down? What's the node's address so I can try to check the status on my own?
Thanks!
